src/main/java/com/palantir/docker/compose/execution/AbstractCommand.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.docker.compose.execution;
+
+import static java.util.stream.Collectors.joining;
+
+import java.io.IOException;
+import java.util.Arrays;
+
+public abstract class AbstractCommand<T extends Executable> {
+
+    private final String commandName;
+    private final AbstractSynchronousExecutable<T> executable;
+
+    public AbstractCommand(String commandName, AbstractSynchronousExecutable<T> executable) {
+        this.commandName = commandName;
+        this.executable = executable;
+    }
+
+    protected final String executeCommand(ErrorHandler errorHandler, String... commands)
```

Comment:
Nit: commands should probably be called arguments.

The way this is used by extending this class and then calling executeCommand is kind of weird - there is no abstract method that children are providing that enables this method to perform the action. So how about using composition in DefaultDockerCompose/Docker? Each of the methods in there would go from executeCommand(errorHandler, commands) to command.execute(errorHandler, commands) which is just as easy to understand when reading those classes but instead this class becomes possible to unit test by itself to ensure the behaviour it has is correct.